pkgs/misc/cups/libppd.nix

```diff
@@ -0,0 +1,56 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, autoreconfHook
+, cups
+, libcupsfilters
+, ghostscript
+, pkg-config
+, poppler_utils
+, mupdf # mutool
+}:
+
+stdenv.mkDerivation rec {
+  pname = "libppd";
+  version = "2.0b2";
+
+  src = fetchFromGitHub {
+    owner = "OpenPrinting";
+    repo = "libppd";
+    rev = "2c650af8c8826a68c0c0293ab5511701eaac019a";
+    hash = "sha256-4KRJfzK/xeWlXmhUOlvGHsMmKqF6R4e5BiIA+h2UTII=";
+  };
+
+  configureFlags = [
+    "--disable-acroread"
+    "--localstatedir=/var"
+    "--sysconfdir=/etc"
+  ];
+
+  nativeBuildInputs = [ autoreconfHook pkg-config ];
+  buildInputs = [
+    cups
+    ghostscript
+    poppler_utils
+    mupdf.bin
+    libcupsfilters
+  ];
+
+  # Fix missing includes on darwin
+  prePatch = lib.strings.optionalString (stdenv.hostPlatform.libc != "glibc")
+    ''
+      substituteInPlace ppd/ppd-collection.cxx \
+        --replace \
+        '#include <sys/wait.h>' \
+        "#include <sys/wait.h>''\n#include <signal.h>''\n#include <libgen.h>''\nextern char **environ;"
+    '';
```

Comment:
have you tried asking upstream yet? better if this is temporary.

Reply:
No, haven't asked yet, but all of my fixes should be upstreamable (just a question of time)